application/src/main/java/org/opentripplanner/routing/algorithm/mapping/StatesToWalkStepsMapper.java

```diff
@@ -515,6 +520,22 @@ private WalkStepBuilder createElevatorWalkStep(State backState, State forwardSta
     return step;
   }
 
+  private WalkStepBuilder createStationEntranceWalkStep(
+    State backState,
+    State forwardState,
+    Edge edge
+  ) {
+    // don't care what came before or comes after
+    var step = createWalkStep(forwardState, backState);
+
+    step.withRelativeDirection(RelativeDirection.CONTINUE);
```

Comment:
Maybe this should be either ENTER_STATION/EXIT_STATION.

Reply:
I discussed this with @HenrikSundell and came to the conclusion that figuring out this information with 100% certainty can be impossible in some cases where the walk leg is between two subway stops.